flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,615 @@
+        if not config.get('api_server_url'):
+            raise plugin.PluginError('Invalid url in config: %s', config)
+
+        api_url = config['api_server_url'] + '/api?' + '&extended=1'
+        if config.get('api_key'):
+            api_url += '&apikey=%s' % config['api_key']
+        if config.get('category_string'):
+            api_url += '&cat=%s' % config['category_string']
+        if 'maxage' in config:
+            maxage = None
+            if config['maxage'] == 'auto':
+                estimator = plugin.get_plugin_by_name('estimate_release').instance
+                est_date = estimator.estimate(entry)
+                if est_date:
+                    if isinstance(est_date, date):
+                        est_date = datetime.combine(est_date, time())
+                    est_date = est_date - timedelta(hours=12)  # add a delta to compensate TZ
+                    maxage = datetime.now() - est_date
+            else:
+                try:
+                    maxage = parse_timedelta(config['maxage'])
+                except ValueError as ex:
+                    log.error('Invalid maxage given in config or estimator: %s', ex)
+
+            if maxage is not None:
+                maxage = max(maxage, timedelta(days=1))  # cap to one day
+                api_url += '&maxage=%s' % maxage.days
+        return api_url
+
+    def parse_from_xml(self, xml_entries):
+        entries = []
+        for xml_entry in xml_entries:
             new_entry = Entry()
+            # skip if we have no link/title
+            if not xml_entry.title or not xml_entry.link:
+                continue
+            # copy xml data to entry
+            new_entry.update_using_map(field_map, xml_entry, ignore_none=True)
+            if not all(key in new_entry for key in field_validation_list):
+                log.warning('Skipping, invalid entry: %s', xml_entry.title)
+                continue
+
+            # fill extra, special data
+            if xml_entry.enclosures:
+                for link in xml_entry.enclosures:
+                    if link.length and link.type == ENCLOSURE_TYPE:
+                        new_entry['content_size'] = int(int(link.length) / 1024 / 1024)  # MB
+            if 'content_size' not in new_entry or new_entry['content_size'] == 0:
+                log.warning('Could not get valid filesize for entry: %s', xml_entry.title)
+            if new_entry.get('newznab_pubdate'):
+                new_entry['newznab_age'] = datetime.now() - new_entry['newznab_pubdate']
 
-            for key in list(rss_entry.keys()):
-                new_entry[key] = rss_entry[key]
-            new_entry['url'] = new_entry['link']
-            if rss_entry.enclosures:
-                size = int(rss_entry.enclosures[0]['length'])  # B
-                new_entry['content_size'] = old_div(size, 2 ** 20)  # MB
             entries.append(new_entry)
+            #_debug_dump_entry(new_entry)  # debug
```

Comment:
This makes no sense to me, sorry. 
So after a PR was merged and i want to add a new feature or fix a bug, i have to write the debug code again? So by this logic i also have to remove the "_add_debug_info" from my sort_by_weight plugin and rewrite it every-time i have to debug a problem?